services/agents/node.go

```diff
@@ -29,6 +29,35 @@ import (
 	"github.com/percona/pmm-managed/utils/collectors"
 )
 
+var v2_26_99 = version.MustParse("2.26.99")
+
+func getExporterFiles(exporter *models.Agent, pmmAgentVersion *version.Parsed) map[string]string {
+	files := exporter.Files()
+
+	// The node exporter prior 2.26 use exporter_shared and gets basic auth config from env.
+	// Starting with pmm 2.27, the exporter uses Prometheus Web Toolkit and needs a config file
+	// with the basic auth users.
+	if pmmAgentVersion.Less(v2_26_99) { // <= 2.27
+		delete(files, "webConfigPlaceholder")
+	}
+	if len(files) == 0 {
+		return nil
+	}
+
+	return files
+}
+
+func getExporterEnv(exporter *models.Agent, pmmAgentVersion *version.Parsed) []string {
+	env := []string{}
+
+	// basic auth via env for older exporters
+	if pmmAgentVersion.Less(v2_26_99) { // <= 2.27
+		env = append(env, fmt.Sprintf("HTTP_AUTH=pmm:%s", exporter.GetAgentPassword()))
+	}
+
+	return env
```

Comment:
What about dont create env var in beginning and in case of <= 2.27 return it immediately and in other cases just return []string{}.

Reply:
simplified